### PR TITLE
offset2D for construction

### LINKED ISSUE #618: forConstruction option in offset2d
I often want to use `Workplane.offset2D` to create construction geometry. For example:

```python
board_dims = (100, 50)
w = (
    cq.Workplane()
    .box(*board_dims, 1)
    .faces(">Z")
    .workplane()
    .rect(*board_dims)
    .offset2D(-2.4)
    .vertices()
    .hole(3)
)
```

Which actually works just fine:
screenshot2021-02-05-112648
Until you go to use `w.ctx.pendingWires`, for example:
```python
board_dims = (100, 50)
w = (
    cq.Workplane()
    .box(*board_dims, 1)
    .faces(">Z")
    .workplane()
    .rect(*board_dims)
    .offset2D(-2.4)
    .vertices()
    .hole(3)
    .faces(">Z")
    .workplane()
    .circle(10)
    .extrude(1)
)
```
screenshot2021-02-05-112923
And now you realise `offset2D` leaves the shape in `pendingWires`.

Is it worth adding a `forConstruction` option, similar to `rect` and other methods?

### PR DESCRIPTION
**Pull request by marcus7070**
Will close #618.

I added a new example that covers offset2D as well.

### CODE REVIEW

File: cadquery/cq.py [hunk 1]
```diff
@@ -3742,22 +3742,31 @@ def toPending(self) -> "Workplane":
...
         rv = list(chain.from_iterable(w.offset2D(d, kind) for w in ws))
 
         self.ctx.pendingEdges = []
-        self.ctx.pendingWires = rv
+        if forConstruction:
+            for wire in rv:
+                wire.forConstruction = True
+            self.ctx.pendingWires = []
+        else:
+            self.ctx.pendingWires = rv
```
Comment (marcus7070) [inline cadquery/cq.py:3769]: Note that this will be cleaned up a bit by https://github.com/CadQuery/cadquery/issues/621#issuecomment-776304825

**Comment by marcus7070** [comment]
Ready for review now.

**Comment by jmwright** [comment]
@marcus7070 Thanks for putting this together. Feel free to request a review or just let us know when this is ready. I see the checks are still running right now.

**Comment by jmwright** [comment]
Whoops, we hit Comment at the same time.

**Review by jmwright — APPROVED** [review]
@marcus7070 I think it looks good, thanks!

We can merge after @adam-urbanczyk has had a chance to review and comment.

**Review by adam-urbanczyk — APPROVED** [review]

**Comment by jmwright** [comment]
@marcus7070 I think once the conflicts are resolved we can merge.

**Comment by jmwright** [comment]
Thanks @marcus7070 !

**Pull request merged**